test/unit/helpers/buyers/applications_helper_test.rb

```diff
@@ -15,6 +15,9 @@ def setup
 
     @fields = { name: 'Field' }
     ProviderDecorator.any_instance.stubs(:application_defined_fields_data).returns([@fields])
+
+    # TODO: Improve component by removing service_plans data when service_plans_visible is false
+    ProviderDecorator.any_instance.stubs(:service_plans_visible).returns(true)
```

Comment:
`any_instance` is a bit dangerous, but I guess we're already drinking from this poison. Ideally, we would control what the whole decoration returns, replacing it with an instance on which these methods could be stubbed.

Reply:
Agreed 100%. I'm changing this helper class in https://github.com/3scale/porta/pull/2459 so I'll improve the tests there.

Reply:
In the end I moved the tests to ProviderDecoratorTests and make them more concise. If they need to be improved we can still do it in the aforementioned PR.